src/java.base/share/classes/jdk/internal/foreign/NativeMemorySegmentImpl.java

```diff
@@ -61,6 +60,13 @@ public Optional<Object> heapBase() {
         return Optional.empty();
     }
 
+    public final long maxByteAlignment() {
+        long alignment = Long.lowestOneBit(address());
+        return alignment == 0
+                ? 1L << 62
+                : alignment;
+    }
```

Comment:
This relies on the fact that `lowestOneBit` returns `0` only if `address()` returns `0`, but I suggest writing this more directly by checking against `address()`:

```suggestion
        return address() == 0
                ? 1L << 62
                : Long.lowestOneBit(address());
```

(I also suggest introducing a name for the `1L << 62` constant, e.g. `MAX_NATIVE_ALIGN`)